react/week3/my-app/src/render.js

```diff
@@ -0,0 +1,189 @@
+import React from "react";
+
+function TodoItem(props) {
+  const isComplete = props.todos.completed;
+  return (
+    <li
+      id="lists"
+      style={{ textDecoration: isComplete ? "line-through" : "none" }}
+    >
+      {props.todos.description} || {props.todos.deadline}
+      <input type="checkbox" onChange={props.handelCheckbox}></input>
+      <button onClick={props.deleteTodo}> Delete</button>
+      <button onClick={props.editTodo}>Edit </button>
+      <button onClick={props.updateTodo}> update </button>
+    </li>
+  );
+}
+
+class TodoList extends React.Component {
+  state = {
+    listOfTodos: [],
+    todoName: "",
+    deadlineDate: "",
+    updateInput: "",
+  };
+
+  addTodo = this.addTodo.bind(this);
+  removeTodo = this.removeTodo.bind(this);
+  handelCheckbox = this.handelCheckbox.bind(this);
+  update = this.update.bind(this);
+
+  feachTodoArray() {
+    return fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    ).then((data) => data.json());
+  }
+
+  addTodo() {
+    this.feachTodoArray().then((response) => {
+      const random = response[Math.floor(Math.random() * response.length)];
+
+      this.setState({
+        listOfTodos: [...this.state.listOfTodos, random],
+      });
+    });
+  }
+
+  addNewTodo(todoName, deadlineDate) {
+    this.setState({
+      listOfTodos: [
+        ...this.state.listOfTodos,
+        {
+          description:
+            todoName === ""
+              ? alert("description can not have an empty value")
+              : todoName,
+          completed: false,
+          deadline:
+            deadlineDate === ""
+              ? alert("deadlindDate can not have an empty value")
+              : deadlineDate,
+        },
+      ],
+    });
+  }
+  update(id) {
+    this.setState({
+      listOfTodos: this.state.listOfTodos.map((task, index) => {
+        if (id === index) {
+          return {
+            ...task,
+            description:
+              this.state.updateInput === ""
+                ? alert("can not update empty value")
+                : this.state.updateInput,
+          };
+        } else {
+          return task;
+        }
+      }),
+    });
+  }
+
+  editTodo(id) {
+    this.setState({
+      listOfTodos: this.state.listOfTodos.map((task, index) => {
+        if (id === index) {
+          return {
+            ...task,
+            description: (
+              <input
```

Comment:
It is not a good idea to mix your data with JSX. Please, consider moving the 'editing' state into the TodoItem component